### ISSUE TITLE
[Cineverse] Unusual characters break download request

### ISSUE THREAD

**Issue by homeofthebadguys** [yt-dlp/yt-dlp#9072]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that yt-dlp is broken on a **supported** site
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

United States

### Provide a description that is worded well enough to be understood

I discovered an issue in the Cineverse extractor where certain characters in a URL can break the extractor, as seen personally with colons in quite a few of Cineverse's titles.

The command line targeted the URL https://www.cineverse.com/details/DMR00008462/Kimagure-Orange-Road:-I-Want-to-Return-to-That-Day

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['-vU', '--skip-download', 'https://www.cineverse.com/details/DMR00008462/Kimagure-Orange-Road:-I-Want-to-Return-to-That-Day']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.12.30 from yt-dlp/yt-dlp [f10589e34] (win_exe)
[debug] Python 3.8.10 (CPython AMD64 64bit) - Windows-10-10.0.19045-SP0 (OpenSSL 1.1.1k  25 Mar 2021)
[debug] exe versions: ffmpeg n6.1.1-1-g61b88b4dda-20240120 (setts), ffprobe n6.1.1-1-g61b88b4dda-20240120
[debug] Optional libraries: Cryptodome-3.19.1, brotli-1.1.0, certifi-2023.11.17, mutagen-1.47.0, requests-2.31.0, sqlite3-3.35.5, urllib3-2.1.0, websockets-12.0
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets
[debug] Loaded 1798 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: stable@2023.12.30 from yt-dlp/yt-dlp
yt-dlp is up to date (stable@2023.12.30 from yt-dlp/yt-dlp)
[CineverseDetails] Extracting URL: https://www.cineverse.com/details/DMR00008462/Kimagure-Orange-Road:-I-Want-to-Return-to-That-Day
[CineverseDetails] DMR00008462: Downloading webpage
[Cineverse] Extracting URL: https://www.cineverse.com/watch/DMR00008462/Kimagure Orange Road: I Want to Return to That Day
[Cineverse] DMR00008462: Downloading webpage
ERROR: DMR00008462: An extractor error has occurred. (caused by KeyError('url')); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
  File "yt_dlp\extractor\common.py", line 718, in extract
  File "yt_dlp\extractor\cineverse.py", line 80, in _real_extract
KeyError: 'url'
```

**Comment by bashonly** [comment]
cc @garret1317

**Comment by garret1317** [comment]
for me it looks like the `idetails` is `{'err_code': 1002, 'err_msg': "You don't have access to this title"}`, despite as you say, everything still being there on the site

I don't have a solution at the moment, but as a workaround, downloading the same episode from retrocrush.tv works: https://www.retrocrush.tv/watch/DMR00008462/Kimagure-Orange-Road:-I-Want-to-Return-to-That-Day

**Comment by garret1317** [comment]
If I pass cookies `--cookies-from-browser firefox` with an account, it starts working
and I get the same error without cookies for https://www.cineverse.com/details/DMR00018764/My-King (randomly selected one that requires login)
so I think the answer is just that they're starting to crack down on accessing stuff without proper auth, and the colons thing was just a coincidence

**Comment by bashonly** [comment]
> for me it looks like the `idetails` is `{'err_code': 1002, 'err_msg': "You don't have access to this title"}`

yeah, this is true of the watch page (`https://www.cineverse.com/watch/DMR00008462/Kimagure Orange Road: I Want to Return to That Day`)

> it looks like the m3u8 url is available from the details page html?

what i meant by this is that the details page (`https://www.cineverse.com/details/DMR00008462/Kimagure-Orange-Road:-I-Want-to-Return-to-That-Day`) contains the m3u8 url even when logged out

json fragment:
```json
"media_hls_url":"https://retrocrush-cineverse.cdn-ak.matchpoint.tv/cinedigm/DMR00008462/62c7308cf1039d14ae278756/hls/master.m3u8","media_mp4_url":"https://cdn.matchpoint.tv/lqy9glmsn5z1srgmxycy3ahe0ou7jtn2n427wtsidlb780yb5dy3ur0un7p3dio3pbc3lrtwy6cc4ummdelpjaqzul8upyj363mz95e9toszzdhu7gu3llq3tblom/4300.mp4","media":[{"url":"https://retrocrush-cineverse.cdn-ak.matchpoint.tv/cinedigm/DMR00008462/62c7308cf1039d14ae278756/hls/master.m3u8","url_clean":"https://retrocrush-cineverse.cdn-ak.matchpoint.tv/cinedigm/DMR00008462/62c7308cf1039d14ae278756/hls/master.m3u8","bitrate":"4300","duration":4265636,"type":"video/vnd.apple.mpegURL","lang":null}]
```

**Comment by john123903** [comment]
> > for me it looks like the `idetails` is `{'err_code': 1002, 'err_msg': "You don't have access to this title"}`
> 
> yeah, this is true of the watch page (`https://www.cineverse.com/watch/DMR00008462/Kimagure Orange Road: I Want to Return to That Day`)
> 
> > it looks like the m3u8 url is available from the details page html?
> 
> what i meant by this is that the details page (`https://www.cineverse.com/details/DMR00008462/Kimagure-Orange-Road:-I-Want-to-Return-to-That-Day`) contains the m3u8 url even when logged out
> 
> json fragment:
> 
> "media_hls_url":"https://retrocrush-cineverse.cdn-ak.matchpoint.tv/cinedigm/DMR00008462/62c7308cf1039d14ae278756/hls/master.m3u8","media_mp4_url":"https://cdn.matchpoint.tv/lqy9glmsn5z1srgmxycy3ahe0ou7jtn2n427wtsidlb780yb5dy3ur0un7p3dio3pbc3lrtwy6cc4ummdelpjaqzul8upyj363mz95e9toszzdhu7gu3llq3tblom/4300.mp4","media":[{"url":"https://retrocrush-cineverse.cdn-ak.matchpoint.tv/cinedigm/DMR00008462/62c7308cf1039d14ae278756/hls/master.m3u8","url_clean":"https://retrocrush-cineverse.cdn-ak.matchpoint.tv/cinedigm/DMR00008462/62c7308cf1039d14ae278756/hls/master.m3u8","bitrate":"4300","duration":4265636,"type":"video/vnd.apple.mpegURL","lang":null}]

is there a solution yet?
if so, can you give me the command to bypass the idetails error?
